src/modules/directory.rs

```diff
@@ -32,8 +32,23 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
         .config_value_i64("fish_style_pwd_dir_length")
         .unwrap_or(FISH_STYLE_PWD_DIR_LENGTH);
 
+    // If use_logical_path is set, read from the envar PWD. This is done by many
+    // logical path tools, including coreutils pwd when run as `pwd -L`. If this
+    // fails, log an error and fall back to the path from context.
+    let use_logical_path = module.config_value_bool("use_logical_path").unwrap_or(true);
+    let current_dir = if use_logical_path {
```

Comment:
I had to change from taking a borrow of the context variable to taking a clone. Here's why: when using the logical path, we read `$PWD` from the environment to construct a PathBuf. Since the PathBuf is created within the scope of the match, it's dropped at the closing brace, so there's no way to return a borrow of it without having the reference outlive the data.

There are two alternate solutions to this:
  - Set up `use_logical_path` as a prompt-wide config which controls how the `context` determines the `current_dir`
  - Declare a mutable outside the match block, set its value within the match, then return a reference to it.

Reply:
Here's another option which doesn't require a copy:

```rust
let pwd_dir = std::env::var("PWD").ok();
let current_dir = pwd_dir
    .as_ref()
    .map(|d| Path::new(d))
    .unwrap_or_else(|| context.current_dir.as_ref());
```

Reply:
Thanks! I had the idea to use something like this while I was developing this thing, but I couldn't quite work out how to get the ownership to work.